### PR TITLE
MMVP exportchannel command

### PR DESCRIPTION
**Pull request by aronasorman**

### CODE REVIEW

File: contentcuration/contentcuration/management/commands/exportchannel.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+    kolibrinode = kolibrimodels.ContentNode.objects.create(
+        title=ccnode.title,
+        pk=ccnode.pk,
+        content_id=ccnode.content_id,
+        description=ccnode.description,
+        sort_order=ccnode.sort_order,
+        license_owner=ccnode.license_owner,
+        kind=ccnode.kind.kind,
+        license_id=1,
```
Comment (jayoshih) [inline contentcuration/contentcuration/management/commands/exportchannel.py]: Need to assign channel's license (set before running command)

File: contentcuration/contentcuration/tests/test_exportchannel.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    fileobj = mixer.blend('contentcuration.File', contentnode=leaf, file_format=ext, preset=videopreset)
+
+    channel = mixer.blend('contentcuration.Channel', main_tree=root, name='testchannel')
+
+    return channel
+
+
+@pytest.mark.django_db
+def test_things_work(channel, settings):
```
Comment (jayoshih) [inline contentcuration/contentcuration/tests/test_exportchannel.py]: add test to make sure channel version is incremented/no nodes have the changed value set to true

File: contentcuration/contentcuration/management/commands/exportchannel.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+class EarlyExit(BaseException):
+    def __init__(self, message, db_path):
+        self.message = message
+        self.db_path = db_path
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('channel_name', type=str)
```
Comment (jayoshih) [inline contentcuration/contentcuration/management/commands/exportchannel.py]: Change to id
Also add license id

File: contentcuration/contentcuration/management/commands/exportchannel.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    def handle(self, *args, **options):
+        try:
+            channel = get_channel_for_name(options["channel_name"])
```
Comment (jayoshih) [inline contentcuration/contentcuration/management/commands/exportchannel.py]: get by id instead

File: contentcuration/contentcuration/management/commands/exportchannel.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+            set_active_content_database('export_staging')
+            # increment the channel version
+            raise_if_nodes_are_all_unchanged(channel)
+            mark_all_nodes_as_changed(channel)
```
Comment (jayoshih) [inline contentcuration/contentcuration/management/commands/exportchannel.py:35]: Add license to nodes

File: contentcuration/contentcuration/models.py [hunk 6]
```diff
@@ -104,7 +104,17 @@ class ContentNode(MPTTModel, models.Model):
...
     """
+    # The id should be the same between the content curation server and Kolibri.
+    id = UUIDField(primary_key=True, default=uuid.uuid4)
+
+    # the content_id is used for tracking a user's interaction with a piece of
+    # content, in the face of possibly many copies of that content. When a user
+    # interacts with a piece of content, all substantially similar pieces of
+    # content should be marked as such as well. We track these "substantially
+    # similar" types of content by having them have the same content_id.
     content_id = UUIDField(primary_key=False, default=uuid.uuid4, editable=False)
```
Comment (jayoshih) [inline contentcuration/contentcuration/models.py]: Still needed?

File: contentcuration/contentcuration/management/commands/exportchannel.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        format = ccfilemodel.file_format
+
+        kolibrifilemodel = kolibrimodels.File.objects.create(
+            pk=ccfilemodel.pk,
+            checksum=ccfilemodel.checksum,
+            extension=format.extension,
+            available=False,
+            file_size=ccfilemodel.file_size,
+            contentnode=kolibrinode,
+            preset=preset,
```
Comment (jayoshih) [inline contentcuration/contentcuration/management/commands/exportchannel.py]: Save pk

File: contentcuration/contentcuration/tests/test_exportchannel.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    # TODO (aron): split different gets/asserts into their own tests
+    call_command('exportchannel', channel.name)
```
Comment (jayoshih) [inline contentcuration/contentcuration/tests/test_exportchannel.py]: will need to change to channel id and license id

**Comment by 66eli77** [comment]
please take note in kolibri, prerequisite and is_related M2M does not use the through middle models anymore. You can directly set the M2M relationship, and be sure to make prerequisite symmetrical=False and is_related symmetrical=True
then you can delete the PrerequisiteContentRelationship and RelatedContentRelationship models to reduce complexity.

File: contentcuration/contentcuration/management/commands/exportchannel.py [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+    channel.main_tree.get_family().update(license_id=license.pk)
+
+
+def map_content_tags(channel):
+    logging.debug("Creating the Kolibri content tags.")
+
+    cctags = ccmodels.ContentTag.objects.filter(
+        channel=channel).values("tag_name")
+    kolibrimodels.ContentTag.objects.bulk_create(
+        [kolibrimodels.ContentTag(**vals) for vals in cctags])
```
Comment (66eli77) [inline contentcuration/contentcuration/management/commands/exportchannel.py:76]: you need to give kolibrimodels.ContentTag a id(UUIDField PK)

Comment (aronasorman) [inline contentcuration/contentcuration/management/commands/exportchannel.py:76]: `kolibri.content.models.ContentTag` still uses the default PK mechanism (`AutoIntegerField`). Should I add in that UUIDField PK then on both content curation and kolibri?

**Comment by aronasorman** [comment]
@66eli77 UUIDField added in https://github.com/jayoshih/content-curation/pull/25/commits/147096ae10173393198eca9e218b003bf1c9a9f2.

**Pull request merged**